src/components/header/header.jsx

```diff
@@ -30,6 +39,17 @@ export const Header = () => {
     openModal(<SignupModal onClose={closeModal} />);
   }
 
+  function openMoviePreviewModal() {
+    openModal(
```

Comment:
There's a (slight) chance that `getRandomMovie` would return us `undefined` and the `MoviePreviewModal` does rely on there being a `selectedMovie` present. Can we call `getRandomMovie` outside of the `openModal` call and only conditionally `openModal` if we get a movie in response?

Reply:
Sure, I've updated the code with that change!